packages/admin/cms-admin/src/generator/future/generateForm/generateFormField.ts

```diff
@@ -283,9 +338,120 @@ export function generateFormField({
 
         const rootQuery = config.rootQuery; //TODO we should infer a default value from the gql schema
         const queryName = `${rootQuery[0].toUpperCase() + rootQuery.substring(1)}Select`;
+        const rootQueryType = findQueryTypeOrThrow(rootQuery, gqlIntrospection);
 
         formFragmentField = `${name} { id ${labelField} }`;
+
+        const filterConfig = config.filter
```

Comment:
@nsams @johnnyomair this pull-request is now dependent on #2372 because it extends the filter-feature introduced with this pull-request. the actual change can be found in this commit: [Implement asyncSelect filter-by-prop feature](https://github.com/vivid-planet/comet/pull/2404/commits/382101214b61c998d12c0a009ad753436d856aad)